Milestone 2

pasivinay opened the pull request:
### **This PR introduces Docker support with a multi-stage Dockerfile for optimized API deployment:**

1. It includes separate build and run stages to ensure efficient image layering and reduce unnecessary dependencies.
3. Environment variables can be injected at runtime to allow flexible configuration without modifying the image.
5. The README file is updated with clear instructions for building the Docker image and running the container.
7. Makefile targets are added to automate Docker operations, simplifying the build and deployment process.
9. Semantic versioning is implemented for image tagging, discouraging the use of the latest tag for better version control.
11. The overall image size is minimized by optimizing layers, reducing unused dependencies, and leveraging smaller base images.

pasivinay committed:
added flask api application scripts

pasivinay committed:
Added environment varible (.env) file

pasivinay committed:
Added requirements.txt file for python packages

pasivinay committed:
Added Makefile and Readme Files

pasivinay committed:
Added postman screenshots

pasivinay committed:
Added postman collection json file

pasivinay committed:
Added gitignore file

pasivinay committed:
Updated readme file

pasivinay committed:
Updated readme file

pasivinay committed:
Fix invalid requests in postman_collection.json file

pasivinay committed:
Fix RESTApi application logging in app.py

pasivinay committed:
Add log files to .gitignore to ignore log files

pasivinay committed:
Add test_api.py file with unit tests for api testing

pasivinay committed:
Update Makefile to add api testing commands

pasivinay committed:
Update README file to add api testing commands

pasivinay committed:
Add pytest packages in python requirements file

pasivinay committed:
Fix loading environment variable file importing dotenv library in app.py

pasivinay committed:
Fix makefile unable to run targets

This commit adds .PHONY to make to makefile to prevent Make from
confusing targets with a file name

pasivinay committed:
Fix formatting of python files

This commit is made to fix formatting of python files using black
formatter.

pasivinay committed:
Fix typo in postman collection json file

This commit fixes typos in request url of two requests in postman collection file

pasivinay committed:
Fix the main README file

This commit is made to make the main project file a guide to subsequent
milestones for the project

pasivinay committed:
Fix over informative README file

This commits removed Postman Demo section making README file simpler.

pasivinay committed:
Add necessary comments to makefile

This commit adds necessary comments to makefile for readability

pasivinay committed:
Add milestone_2 folder copied from milestone_1

This commit adds a copy of milestone_1 folder named as milestone_2
for continuing working on milestone 2. Further development of
milestone 2 will be done under this folder.

pasivinay committed:
Remove postman snapshots

This commit removes postman snapshot as no longer required.

pasivinay committed:
Add Dockerfile with builder stage

This commit adds a Dockerfile with a builder stage. This stage installs
all dependencies required for the application using requirements.txt
file. It also copies application files to working directory.

pasivinay committed:
Add runtime stage to dockerfile

Adds a runtime stage to dockerfile for deploying the flask api. This
stage copies artifacts from builder stage required for runtime stage.

pasivinay committed:
Update readme file

Adds and describes the purpose of milestone 2 in main readme file.

pasivinay committed:
Modify makefile for docker commands

This commit modifies Makefile to add targets for docker commands to build,
run and push docker image for flask api.

pasivinay committed:
Modify Makefile to add and modify targets

This commit adds target for migrate and modifies tag-push target to
fetch dockerhub username dynamically.

pasivinay committed:
Modify README file for docker instruction

Adds instructions to README file for steps to build, tag, push and run
docker image. It also updates the pre-requisites for milestone 2.

pasivinay committed:
Modify README Project title

Modified the README project title for milestone 2

pasivinay committed:
Removed .env file and add an entry in .gitignore

This commit removes the .env file and add an entry for the same in
.gitignore file

pasivinay committed:
Modify README for milestone 2

This commit makes changes to instructions in README file suitable for
milestone 2

pasivinay committed:
Add example env file

This commit adds an example .env file for reference

pasivinay committed:
Modify makefile to add lint target

This commit modifies makefile to add lint target to makefile and add
missing targets from .PHONY Keyword.

rajeshecb70 commented on milestone_2/Dockerfile:
In Dockerfile add only specific project files not all files.

pasivinay replied:
Changes done:
1. Restructured application files and added them to dedicated app folder
2. Modified Dockerfile to copy files from app folder only.

pasivinay committed:
Restructure application files

This commit adds all application file to app folder for optmized usage.

pasivinay committed:
Modify dockerfile to copy application files only

This commit will modify dockerfile to copy only application files to
docker images. These application files are stored in app folder.

rajeshecb70 reviewed (commented):
Why is milestone-1 content showing here? Please check. Always create a new branch from the master.
